src/syscall/syscall_libc.go

```diff
@@ -51,11 +51,30 @@ func Open(path string, flag int, mode uint32) (fd int, err error) {
 }
 
 func Mkdir(path string, mode uint32) (err error) {
-	return ENOSYS // TODO
+	data := append([]byte(path), 0)
```

Comment:
I see that existing code also does this.
If you want to improve on that, you could create a new helper function that creates a C string from a Go string with only a single allocation (make a []byte slice of one byte larger and copy the string into it).

Reply:
done.